### ISSUE TITLE
Ampersands in predicate literals

### ISSUE THREAD

**Issue by jimhol** [StevenLooman/saxpath#16]
I don't seem to be able to find the way to escape these. Any use of ampersand, even when trying to prefix an entity such as &amp;amp;, fails with

> SyntaxError: Expected "\"" or [a-z0-9_ ]i but "&" found.

Ditto with "%". Am I missing the trick or is this not supported?

**Comment by StevenLooman** [comment]
At this moment this isn't supported. The grammar used to parse the XPath
expression has to be expanded to support this. I don't foresee any problems
in doing this. You can do this yourself by expanding the file
`lib/xpath_parser.pegjs` and generating the XPath expression parser. The
parser can be generated by running: `make lib/xpath_parser.js`

If you plan on fixing this, please do write a unit test for this. If not,
I'll see what I can do in the future.

**Comment by StevenLooman** [comment]
@jimhol Do you have an example?

**Comment by jimhol** [comment]
Sorry. My requirements changed under me, to something that could not be supported by this anyway. I didn't mean to completely abandon this however! It seems my experiments did not get checked in and are not on this machine, and I'll next be on that other machine next year. This is not going well. Let me try to remember.

Essentially I had XML containing an element with an string attribute containing an ampersand, such as:

```<book title="Pride &amp; Prejudice">...</book>```

So I was trying things like "//book@title='Pride & Prejudice'". The title string I needed to find the element for was coming from another source. I was looking into how to escape it programmatically and I was testing possibilities this, but only because it wasn't working when I tried it raw. I hadn't spotted that it wasn't in the grammar. I suppose it could work raw since we're not expressing the XPath within XML.
